tests/test_special_ops.py

```diff
@@ -254,3 +256,40 @@ def test_accuracy_resolve_conj(shape, dtype):
     with flag_gems.use_gems():
         z = y.resolve_conj()
     assert not z.is_conj()
+
+
+@pytest.mark.parametrize("shape", UT_SHAPES_1D + UT_SHAPES_2D)
+@pytest.mark.parametrize("dtype", FLOAT_DTYPES)
+@pytest.mark.parametrize("n_samples", [1000])
+def test_accuracy_multinomial_with_replacement(shape, dtype, n_samples):
+    if shape[-1] == 1:
+        dist = torch.rand(size=shape, dtype=dtype, device="cuda")
```

Comment:
tests on cpu need to_reference

Reply:
In this test, we don't need a reference result. It's device only test.